### PR TITLE
Cover more test/widgets tests with leak tracking #12.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
### Description
This PR changes `testWidgets` by `testWidgetsWithLeakTracking` in number of `flutter/test/widgets` tests according to the https://github.com/dart-lang/leak_tracker/issues/134.

### Tests
- Updates `animated_size_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `framework_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `html_element_view_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `list_wheel_scroll_view_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `navigator_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `overlay_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `router_restoration_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `scrollable_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `transform_test.dart` to use `testWidgetsWithLeakTracking`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by polina-c** [comment]
LGTM
Will formally approve and merge it after merging this change to leak_tracker, and upgrading Flutter: https://github.com/dart-lang/leak_tracker/pull/149

**Review by polina-c — APPROVED** [review]

**Comment by polina-c** [comment]
@ksokolovskyi , may I ask you to resolve conflicts?

**Comment by ksokolovskyi** [comment]
@polina-c, thanks a lot for the review!

**Pull request merged**
